registry/storage/driver/s3/s3.go

```diff
@@ -587,6 +587,11 @@ func (d *driver) List(path string) ([]string, error) {
 	if path != "/" && path[len(path)-1] != '/' {
 		path = path + "/"
 	}
+	prefix := ""
```

Comment:
Shouldn't this be encapsulated in whatever function converts to and from s3 paths?